### DISCUSSION TITLE
stumpy.motifs step size

### DISCUSSION THREAD

**Discussion by marwaajouz** [TDAmeritrade/stumpy#903] [Q&A]
When using stumpy.motifs to get motifs of a time series, we specify a window size; however, I can not find a way to specify the step size, meaning, if I have window size = 90, the function takes this window size as input to specify the length of one sequence of motif but still goes through each point to search for a motif with step size = 1. I am getting overlapping motifs as output.

<img width="632" alt="image" src="https://github.com/TDAmeritrade/stumpy/assets/135465693/d52a50ce-63d7-48c6-8fc0-39924509907e">

How could I specify that I need step size = window size so that I do not get overlapping motifs as a result?

**Comment by seanlaw** [comment]
@marwaajouz Thank you for your question and welcome to the STUMPY community. So, there is no way to specify a "step size" when searching for each subsequent motif. This is by design so as not to make any assumptions about what the user wants/expects (since everyone wants/expects different things). Instead, you should post-process the motifs that are returned to you (which are sorted in order of "subsequence with the closest matches" to the "subsequence with the farthest matches". This should be a simple for-loop that loops over each motif idx (i.e., `motif_indices[:, 0]`) in order and then eliminate the ones that overlap with an existing motif:

```
# This had NOT been tested but I hope you can get the main point
motif_distances, motif_indices = stumpy.motifs(T, mp[:, 0])

valid_indices = [motif_indices[0, 0]]
for candidate_idx in motif_indices[1:, 0]:
    append_idx = True
    for valid_idx in valid_indices:
        if valid_idx - m < candidate_idx < valid_idx + m:
            append_idx = False
            break
    if append_idx:
        valid_indices.append(candidate_idx)

mask = np.isin(motif_indices[:, 0], valid_indices)
no_overlap_motif_distances = motif_distances[mask]
no_overlap_motif_indices = motif_indices[mask]
```

I hope that answers your question
